File: packages/core-dto/src/main/kotlin/com/vander/core/dto/SpotifyPlaylistDto.kt
package com.vander.core.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SpotifyPlaylistDto(
    val collaborative: Boolean,
    val description: String,
    @SerialName("external_urls") val externalUrls: ExternalUrlsDto,
    val href: String,
    val id: String,
    val images: List<SpotifyImageDto>,
    val name: String,
    val owner: SpotifyOwnerDto,
    val public: Boolean? = null,
    @SerialName("snapshot_id") val snapshotId: String,
    val tracks: SpotifyTracksDto,
    val type: String,
    val uri: String
)

@Serializable
data class ExternalUrlsDto(
    val spotify: String
)

@Serializable
data class SpotifyOwnerDto(
    @SerialName("external_urls") val externalUrls: ExternalUrlsDto,
    val href: String,
    val id: String,
    val type: String,
    val uri: String,
    @SerialName("display_name") val displayName: String
)

@Serializable
data class SpotifyTracksDto(
    val href: String,
    val total: Int
)

@Serializable
data class SpotifyImageDto(
    val url: String,
    val height: Int? = null,
    val width: Int? = null
)

File: packages/core-dto-ts/src/lib/spotify/SpotifyPlaylistDto.ts
import { ExternalUrlsDto } from './ExternalUrlsDto';
import { SpotifyImageDto } from './SpotifyImageDto';
import { SpotifyOwnerDto } from './SpotifyOwnerDto';
import { SpotifyTracksDto } from './SpotifyTracksDto';

export interface SpotifyPlaylistDto {
  collaborative: boolean;
  description: string;
  external_urls: ExternalUrlsDto;
  href: string;
  id: string;
  images: SpotifyImageDto[];
  name: string;
  owner: SpotifyOwnerDto;
  public?: boolean; // nullable en Kotlin
  snapshot_id: string;
  tracks: SpotifyTracksDto;
  type: string;
  uri: string;
}
